### DISCUSSION TITLE
slidingwindowinferer for WSI

### DISCUSSION THREAD

**Discussion by shubhaminnani** [Project-MONAI/tutorials#1682] [Q&A] [answered]
Hi,
I am trying to infer WSI in .svs format at 40x magnification, will slidingwindowinferer support WSI with patch size of 256x256?

Thanks,
Shubham

**Comment by KumoLiu** [comment]
Hi @drbeh, could you please help share some comments on this one?
Thanks in advance!

**Comment by bhashemian** [comment, marked as answer]
Hi @shubhaminnani, are you referring to [`SlidingWindowInferer`](https://github.com/Project-MONAI/MONAI/blame/87152d106557bdc37de0ee923be4e9a25c4c003c/monai/inferers/inferer.py#L382) in MONAI? If so, it should work with any patch size as it is has `roi_size` as an input parameter.

Alternatively, you can use [`PatchInferer`](https://github.com/Project-MONAI/MONAI/blame/87152d106557bdc37de0ee923be4e9a25c4c003c/monai/inferers/inferer.py#L83) along with  [`SlidingWindlowSplitter`](https://github.com/Project-MONAI/MONAI/blob/87152d106557bdc37de0ee923be4e9a25c4c003c/monai/inferers/splitter.py#L94) and [`AvgMerger`](https://github.com/Project-MONAI/MONAI/blob/87152d106557bdc37de0ee923be4e9a25c4c003c/monai/inferers/merger.py#L92) to achieve what you mentioned. 
Here you can find some examples for WSI at the end of this tutorial: https://github.com/Project-MONAI/tutorials/blob/main/patch_inferer/modular_patch_inferer.ipynb

**Reply by shubhaminnani** [reply]
Thank you @drbeh for confirmation of using WSI in .svs format with SlidingWindowInferer. I will check and let you know if any issues.

**Reply by bhashemian** [reply]
@shubhaminnani, SlidingWindowInferer's input is a tensor already loaded so you need to use WSIReader to load your svs file first (if it fits in memory). Otherwise, use the alternative that I mentioned.

**Reply by shubhaminnani** [reply]
Sure, Thank you!
